frontend/packages/dev-console/src/components/import/deployImage-validation-utils.ts

```diff
@@ -12,8 +13,11 @@ export const deployValidationSchema = yup.object().shape({
     name: yup.string().required('Required'),
   }),
   application: yup.object().shape({
-    name: yup.string().required('Required'),
-    selectedKey: yup.string().required('Required'),
+    selectedKey: yup.string(),
+    name: yup.string().when('selectedKey', {
+      is: (selectedKey) => selectedKey === CREATE_APPLICATION_KEY,
```

Comment:
I think you can just do `is: CREATE_APPLICATION_KEY` now that you're doing simple equality.